src/Components/build/repo.props

```
@@ -0,0 +1,8 @@
+<Project>
+  <Import Project="..\..\..\build\dependencies.props" />
+  <Import Project="..\..\..\build\runtimes.props" />
+  <PropertyGroup>
+    <!-- TODO: temporary while we reorganize source code and refactor dependency management -->
+    <DisablePackageReferenceRestrictions>true</DisablePackageReferenceRestrictions>
```

```suggestion
    <DisablePackageReferenceRestrictions>true</DisablePackageReferenceRestrictions>
    <BuildSolutions>false</BuildSolutions>
    <Projects>$(MSBuildThisFileDirectory)..\**\*.csproj</Projects>
```

If you also add BuildSolutions, you don't have to use the top-level src/ test/ layout, and can organize your projects like we have started doing in aspnet/Extensions.

```
src/
  Components/
     src/
       Microsoft.AspNetCore.Components.csproj
     test/
       Microsoft.AspNetCore.Components.Tests.csproj
```

---

How this would work for functional tests?
```
  Components/
     src/
       Microsoft.AspNetCore.Components.csproj
     test/
       Microsoft.AspNetCore.Components.Tests.csproj
     functionaltests\
       Microsoft.AspNetCore.Components.FunctionalTests.csproj
```

---

I have a slight preference for something like this. @Eilon care to chime in on src code organization patterns? This will set precedence for how we do things as we begin merging new stuff.
```
  Components/
     src/
       Microsoft.AspNetCore.Components.csproj
     test/
       UnitTests/
            Microsoft.AspNetCore.Components.Tests.csproj
       FunctionalTests/
            Microsoft.AspNetCore.Components.FunctionalTests.csproj
```

---

I like having just 1 `src` and 1 `test`, though I'm not sure if there is a need for sub-folders beyond that. What is the value of the extra sub-folders?

---

hmm, this kind of doesn't work for functional tests in the sense that normally they are not tied to a single assembly.

---

Not sure I understand - what doesn't work?

---

Based on the pattern on DataProtection I would favor something like this
```
src\
  Components\
    Components\
      src\
        Components.csproj
      test\
        Components.Tests.csproj
    Components.Mvc\
      src\
        Components.Mvc.csproj
      test\
        Components.Mvc.Tests.csproj
    functionaltests\
     Components.FunctionalTests.csproj
```

With functional tests being a top level subfolder of the top level area folder (components, dataprotection,logging, configuration, or similar area)

---

> Not sure I understand - what doesn't work?

In MVC's example, there isn't a 1 to 1 mapping of implementation and test assemblies.

What we've done in aspnet/Extensions for this case is similar to @javiercn's last suggestion. See https://github.com/aspnet/Extensions/tree/master/src/Configuration for example.

```
- src/Stuff/
   + One/src/
   + Two/src/
   - test/
     +  OneAndTest.FunctionalTests.csproj
```

---

I'm fine with that.

---

Chat with @natemcmaster  offline, we settled for his suggestion. Which is essentially the same as my proposal except for functionaltests->tests